src/gui/folderstatusmodel.cpp

```diff
@@ -863,69 +873,68 @@ void FolderStatusModel::slotApplySelectiveSync()
     resetFolders();
 }
 
-void FolderStatusModel::slotSetProgress(const ProgressInfo &progress)
+void FolderStatusModel::slotSetProgress(const ProgressInfo &progress, Folder *f)
 {
-    auto par = qobject_cast<QWidget *>(QObject::parent());
-    if (!par->isVisible()) {
+    if (!qobject_cast<QWidget *>(QObject::parent())->isVisible()) {
         return; // for https://github.com/owncloud/client/issues/2648#issuecomment-71377909
     }
 
-    Folder *f = qobject_cast<Folder *>(sender());
-    if (!f) {
-        return;
-    }
-
-    int folderIndex = -1;
-    for (int i = 0; i < _folders.count(); ++i) {
-        if (_folders.at(i)._folder == f) {
-            folderIndex = i;
-            break;
+    const int folderIndex = [f, this] {
+        int folderIndex = -1;
+        for (int i = 0; i < _folders.count(); ++i) {
+            if (_folders.at(i)._folder == f) {
+                folderIndex = i;
+                break;
+            }
         }
-    }
+        return folderIndex;
+    }();
     if (folderIndex < 0) {
         return;
     }
 
     auto *pi = &_folders[folderIndex]._progress;
 
-    QVector<int> roles;
-    roles << FolderStatusDelegate::SyncProgressItemString
-          << FolderStatusDelegate::WarningCount
-          << Qt::ToolTipRole;
+    const QVector<int> roles = { FolderStatusDelegate::SyncProgressItemString, FolderStatusDelegate::WarningCount, Qt::ToolTipRole };
 
     if (progress.status() == ProgressInfo::Discovery) {
         if (!progress._currentDiscoveredRemoteFolder.isEmpty()) {
             pi->_overallSyncString = tr("Checking for changes in remote '%1'").arg(progress._currentDiscoveredRemoteFolder);
             emit dataChanged(index(folderIndex), index(folderIndex), roles);
```

Comment:
I wonder if we should not check if `progress.status()` is different than the status that is stored currently. If it is **not** different we could save the `dataChanged()` emit completely, or not?